certification/internal/k8s/openshift.go

```diff
@@ -0,0 +1,286 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	client "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/client"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+)
+
+type OpenshiftCLIEngine struct{}
+
+func (pe OpenshiftCLIEngine) CreateNamespace(name string, opts cli.OpenShiftCliOptions, config *rest.Config) (*cli.OpenshiftCreateReport, error) {
+
+	k8sClientset, err := kubernetes.NewForConfig(config)
```

Comment:
Can we instead add an accessor method like `OpenshiftCLIEngine.New()` that accepts the configuration necessary  to build out a client, and then store that client in an internal struct key (or similar)?